common/strings.c

```diff
@@ -293,3 +293,41 @@ TDNFReplaceString(
     TDNF_SAFE_FREE_MEMORY(pszTempReplace);
     goto cleanup;
 }
+
+uint32_t
+TDNFTrimSuffix(
+    char* pszSource,
+    const char* pszSuffix
+    )
+{
+    uint32_t dwError = 0;
+    int nSourceStrLen = 0, nSuffixStrLen = 0;
+
+    if(!pszSource || !pszSuffix)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    nSourceStrLen = strlen(pszSource);
+    nSuffixStrLen = strlen(pszSuffix);
+
+    if(nSuffixStrLen > nSourceStrLen)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    while(nSuffixStrLen > 0 &&
```

Comment:
Is it better to check nSourceStrLen !=0 and nSuffixStrLen != 0 before going to this loop?

Reply:
That case is implicit. It won't enter the loop in both cases.

Reply:
Thanks right @cbsiddharth